src/server.ts

```diff
@@ -145,134 +67,11 @@ export class Server extends events.EventEmitter {
-		return this.__waitForServerUp()
-			.timeout(PROCESS_TIMEOUT, `Couldn't start Pact with PID: ${this.__instance.pid}`)
-			.then(() => {
-				this.__running = true;
-				this.emit(Server.Events.START_EVENT, this);
-				return this;
-			});
-	}
-
-	// Stop the server instance, no more mocking
-	public stop(): q.Promise<Server> {
-		const pid = this.__instance ? this.__instance.pid : -1;
-		return q(pactUtil.killBinary(this.__instance))
-			.then(() => this.__waitForServerDown())
-			.timeout(PROCESS_TIMEOUT, `Couldn't stop Pact with PID '${pid}'`)
-			.then(() => {
-				this.__running = false;
-				this.emit(Server.Events.STOP_EVENT, this);
-				return this;
-			});
-	}
-
-	// Deletes this server instance and emit an event
-	public delete(): q.Promise<Server> {
-		return this.stop().tap(() => this.emit(Server.Events.DELETE_EVENT, this));
-	}
-
-	// Wait for pact-mock-service to be initialized and ready
-	private __waitForServerUp(): q.Promise<any> {
-		let amount = 0;
-		const deferred = q.defer();
-
-		const retry = () => {
-			if (amount >= RETRY_AMOUNT) {
-				deferred.reject(new Error("Pact startup failed; tried calling service 10 times with no result."));
-			}
-			setTimeout(check.bind(this), CHECKTIME);
-		};
-
-		const check = () => {
-			amount++;
-			if (this.options.port) {
-				this.__call(this.options).then(() => deferred.resolve(), retry.bind(this));
-			} else {
-				retry();
-			}
-		};
-
-		check(); // Check first time, start polling
-		return deferred.promise;
-	}
-
-	private __waitForServerDown(): q.Promise<any> {
-		let amount = 0;
-		const deferred = q.defer();
-
-		const check = () => {
-			amount++;
-			if (this.options.port) {
-				this.__call(this.options).then(() => {
-					if (amount >= RETRY_AMOUNT) {
-						deferred.reject(new Error("Pact stop failed; tried calling service 10 times with no result."));
-						return;
-					}
-					setTimeout(check, CHECKTIME);
-				}, () => deferred.resolve());
-			} else {
-				deferred.resolve();
-			}
-		};
-
-		check(); // Check first time, start polling
-		return deferred.promise;
-	}
-
-	private __call(options: ServerOptions): q.Promise<any> {
-		const deferred = q.defer();
-		const config: any = {
-			uri: `http${options.ssl ? "s" : ""}://${options.host}:${options.port}`,
-			method: "GET",
-			headers: {
-				"X-Pact-Mock-Service": true,
-				"Content-Type": "application/json"
-			}
-		};
-
-		if (options.ssl) {
-			process.env.NODE_TLS_REJECT_UNAUTHORIZED = "0";
-			config.agentOptions = {};
-			config.agentOptions.rejectUnauthorized = false;
-			config.agentOptions.requestCert = false;
-			config.agentOptions.agent = false;
-		}
-
-		http(config, (err: any, res: any) => (!err && res.statusCode === 200) ? deferred.resolve() : deferred.reject(`HTTP Error: '${JSON.stringify(err ? err : res)}'`));
-
-		return deferred.promise;
+	protected spawnBinary(): ChildProcess {
+		return pactUtil.spawnBinary(`${pact.mockServicePath} service`, this.options, this.__argMapping);
```

Comment:
I feel like this could be abstracted further?

Reply:
Arguably. I thought about a few options but nothing felt 'good'. I landed on this as I felt it was the most clear, but if you had thoughts on a nice way I'm keen to hear.

Reply:
well, they all do exactly the same thing except for the first argument.  Maybe this is something else that can go in the AbstractService constructor?

Reply:
Yeah, I thought that but felt wasn't ideal. Second opinion is always good.

I've pushed up an example, let me know what you think.